test/demo/utils/fdc3RemoteExecution.ts

```diff
@@ -99,19 +96,25 @@ export async function addContextListener(executionTarget: Identity): Promise<Rem
     return createRemoteContextListener(executionTarget, id);
 }
 
-export async function addIntentListener(executionTarget: Identity, intent: IntentType): Promise<RemoteIntentListener> {
-    const id = await ofBrowser.executeOnWindow(executionTarget, function(this:TestWindowContext, intentRemote: IntentType): number {
+export async function addIntentListener(executionTarget: Identity, intent: IntentType, listener?: (context: Context) => any): Promise<RemoteIntentListener> {
+    const remoteFn = listener ? await ofBrowser.getOrMountRemoteFunction(executionTarget, listener) : undefined;
+    const id = await ofBrowser.executeOnWindow(executionTarget, function(this: TestWindowContext, intentRemote: IntentType, listenerRemote?: any): number {
         if (this.intentListeners[intentRemote] === undefined) {
             this.intentListeners[intentRemote] = [];
         }
         const listenerID = this.intentListeners[intentRemote].length;
-        this.intentListeners[intentRemote][listenerID] = this.fdc3.addIntentListener(intentRemote, (context) => {
+        this.intentListeners[intentRemote][listenerID] = this.fdc3.addIntentListener(intentRemote, async (context) => {
             this.receivedIntents.push({listenerID, intent: intentRemote, context});
+            // Catch and throw so it rejects correctly client side
+            const result = listenerRemote ? await listenerRemote(context).catch(() => {
+                throw new Error();
```

Comment:
Should this guy populate the error with what comes from the catch?

Reply:
The caught error is thrown now.